models/models_methods.go

```diff
@@ -232,45 +232,46 @@ func (step *StepModel) FillMissingDefaults() error {
 
 // IsStepExist ...
 func (collection StepCollectionModel) IsStepExist(id, version string) bool {
-	_, found := collection.GetStep(id, version)
-	return found
+	_, stepFound, versionFound := collection.GetStep(id, version)
+	return (stepFound && versionFound)
 }
 
 // GetStep ...
-func (collection StepCollectionModel) GetStep(id, version string) (StepModel, bool) {
-	stepVer, isFound := collection.GetStepVersion(id, version)
-	return stepVer.Step, isFound
+func (collection StepCollectionModel) GetStep(id, version string) (StepModel, bool, bool) {
+	stepVer, isStepFound, isVersionFound := collection.GetStepVersion(id, version)
+	return stepVer.Step, isStepFound, isVersionFound
 }
 
 // GetStepVersion ...
-func (collection StepCollectionModel) GetStepVersion(id, version string) (StepVersionModel, bool) {
+func (collection StepCollectionModel) GetStepVersion(id, version string) (StepVersionModel, bool, bool) {
 	stepHash := collection.Steps
-	stepVersions, found := stepHash[id]
-	if !found {
-		return StepVersionModel{}, false
-	}
+	stepVersions, stepFound := stepHash[id]
```

Comment:
if `stepHash` does not contain `id` -> `stepVersions` may will be nil

handle `stepFound` asap